nslsii/iocs/eps_two_state_ioc_sim.py

```diff
@@ -2,6 +2,25 @@
 from caproto.server import pvproperty, PVGroup
 from caproto.server import template_arg_parser, run
 from caproto import ChannelType
+import contextvars
+import functools
+
+internal_process = contextvars.ContextVar('internal_process',
+                                          default=False)
+
+
+def no_reentry(func):
```

Comment:
I recall some discussion in caproto that, IIRC, this does not do what you intend. But since I could be wrong and this is just a IOC supporting a test, I will not hold up the PR over it.

Reply:
This prevents the validator from being entered a second time in the same task which is exactly what we want here (or the internal write will infinitely recourse).

Reply:
I think "task" is not what you think it is here, but I'll pick this up later.